emcc.py

```diff
@@ -93,6 +95,7 @@
     # wasm-ld doesn't support soname or other dynamic linking flags (yet).   Ignore them
     # in order to aid build systems that want to pass these flags.
     '-soname': True,
+    '--allow-shlib-undefined': False,
```

Comment:
This doesn't catch -allow-shlib-undefined (with a single dash), which lld also accepts. The single-dash argument should also be added. Alternatively, flags could also be matched against this list as both single and double dash variants.

Reply:
Do you have a build system that insists on passing `-allow-shlib-undefined`.   Its almost always better to stop passing the flag erroneously if you can.  

For example, for some of these flags it seems that autoconf insists on passing them so blanket ignoring them is a net win.   In other cases we push back and ask people to modify the build rules.

Reply:
The buildsystem is cmake, and in my use case not really under my control.

But since we're already ignoring `allow-shlib-undefined`, we probably should ignore all forms of specifying it.

Reply:
Sure, I don't object to adding this.

I'm just also keen to fix any bad assumptions present in upstream builds systems as well. 

Feel free upload a pr to handle the other case (if its you are not setup to do that easily one of us can).

Reply:
I can do it.

Reply:
https://github.com/emscripten-core/emscripten/pull/11267